### PR TITLE
feat(store): add life cycle interface and test

### LINKED ISSUE #272: Interfaces for Livecycle Events
I think it would be good to create a simple Interface that needs to be implemented on the state class for the livecycle events.

Implementing the interface would be more intuitive than guessing the signature of the livecycle method.

`implements NgxsOnInit`

### PR DESCRIPTION
**Pull request by deebloo**
closes #272

### CODE REVIEW

File: packages/store/src/symbols.ts [hunk 1]
```diff
@@ -63,3 +63,9 @@ export interface ActionOptions {
    */
   cancelUncompleted: boolean;
 }
+
+export interface NgxsOnInit {
+  ngxsOnInit(ctx?: StateContext<any>): void | any;
+}
+
+export type NgxsLifeCycle = Partial<NgxsOnInit>;
```
Comment (deebloo) [inline packages/store/src/symbols.ts]: this makes sure that we know internally that lifecycle methods are optional. we can also add additional life cycle interfaces like this:

```TS
export type NgxsLifeCycle = Partial<NgxsOnInit, NgxsOnDestroy>;
```

Comment (amcdnl) [reply packages/store/src/symbols.ts]: when would destroy ever happen?

Comment (deebloo) [reply packages/store/src/symbols.ts]: oh i'm not saying it would. I am just showing an example of how we can compose the lifecycle in a typesafe way

**Comment by yarrgh** [comment]
What about having a generic interface as well? That way the StateContext can be correctly typed.

**Pull request merged**